packages/components/Button/src/useButton.ts

```diff
@@ -1,30 +1,28 @@
 import * as React from 'react';
-import { useAsPressable, useKeyProps, useOnPressWithFocus, useViewCommandFocus } from '@fluentui-react-native/interactive-hooks';
-import { ButtonProps, ButtonState } from './Button.types';
-import { memoize } from '@fluentui-react-native/framework';
-import { AccessibilityState } from 'react-native';
+import { usePressableState, useKeyProps, useOnPressWithFocus, useViewCommandFocus } from '@fluentui-react-native/interactive-hooks';
+import { ButtonProps, ButtonInfo } from './Button.types';
 
-export const useButton = (props: ButtonProps): ButtonState => {
+export const useButton = (props: ButtonProps): ButtonInfo => {
   // attach the pressable state handlers
   const defaultComponentRef = React.useRef(null);
-  const { onClick, accessibilityState, componentRef = defaultComponentRef, disabled, loading, enableFocusRing, focusable, ...rest } = props;
+  const { onClick, componentRef = defaultComponentRef, disabled, loading, enableFocusRing, focusable, ...rest } = props;
   const isDisabled = !!disabled || !!loading;
   // GH #1336: Set focusRef to null if button is disabled to prevent getting keyboard focus.
   const focusRef = isDisabled ? null : componentRef;
   const onClickWithFocus = useOnPressWithFocus(focusRef, onClick);
-  const pressable = useAsPressable({ ...rest, disabled: isDisabled, onPress: onClickWithFocus });
+  const pressable = usePressableState({ ...rest, onPress: onClickWithFocus });
   const onKeyUpProps = useKeyProps(onClick, ' ', 'Enter');
   const hasTogglePattern = props.accessibilityActions && !!props.accessibilityActions.find((action) => action.name === 'Toggle');
 
   return {
     props: {
       ...onKeyUpProps,
       ...pressable.props, // allow user key events to override those set by us
+      ...(isDisabled && { disabled: isDisabled }), // Due to a bug in React Native, unconditionally passing this may cause unnecessary re-renders. Therefore, let's only pass it in if it's defined.
```

Comment:
https://github.com/facebook/react-native/blob/b55518c4bf22ea6f409b58151d9c84fa161d6e02/Libraries/Components/Pressable/Pressable.js#L206

If `disabled` is passed, then a new object is created, which causes unnecessary renders. I think I'll open an issue on React Native Core so I can reference it here.